stl/inc/xmemory

```diff
@@ -2083,34 +2083,40 @@ _NODISCARD _CONSTEXPR20 _FwdIt remove_if(_FwdIt _First, const _FwdIt _Last, _Pr
 
 // FUNCTION TEMPLATE _Erase_remove
 template <class _Container, class _Uty>
-void _Erase_remove(_Container& _Cont, const _Uty& _Val) { // erase each element matching _Val
+typename _Container::size_type _Erase_remove(_Container& _Cont, const _Uty& _Val) { // erase each element matching _Val
     auto _First      = _Cont.begin();
     const auto _Last = _Cont.end();
     _Seek_wrapped(_First, _STD remove(_Get_unwrapped(_First), _Get_unwrapped(_Last), _Val));
+    const auto _Removed = _Idl_distance<decltype(_First)>(_Get_unwrapped(_First), _Get_unwrapped(_Last));

```

Comment:
_Idl_distance is the "best effort guess" at size that can return _Distance_unknown. You should probably be comparing container size() like you did elsewhere.

Reply:
Ah I see, I was just following the paper that used `std::distance` for everything but map/set, figured there was a reason for that.  
I saw uses of `_Idl_distance ` so I assumed that was to be used and I ended up with the correct results with it.